src/lib/yang/schema.lua

```diff
@@ -795,20 +795,28 @@ function resolve(schema, features)
       end
       if node.body then
          node.body = shallow_copy(node.body)
-         for k,v in pairs(node.body or {}) do
```

Comment:
I think we should only visit each k,v in node.body one time.  Given that we need to augment node.body as we visit, we should build up the result in a separate table.  We assert as we add into that result table that the identifier is unique.  We do not need to revisit the result of a grouping lookup: the lookup_lazy will handle that for us.  WDYT?